modules/apps/content-dashboard/content-dashboard-document-library-test/src/testIntegration/java/com/liferay/content/dashboard/document/library/internal/item/action/provider/test/PreviewFileEntryContentDashboardItemActionProviderTest.java

```diff
@@ -0,0 +1,262 @@
+		Assert.assertTrue(
+			url.contains(
+				"mvcRenderCommandName=%2Fdocument_library%2Fview_file_entry"));
+	}
+
+	@Test
+	public void testGetKey() {
+		Assert.assertEquals(
+			"preview", _contentDashboardItemActionProvider.getKey());
+	}
+
+	@Test
+	public void testGetType() {
+		Assert.assertEquals(
+			ContentDashboardItemAction.Type.PREVIEW,
+			_contentDashboardItemActionProvider.getType());
+	}
+
+	@Test
+	public void testIsShow() throws Exception {
+		_fileEntry = _dlAppLocalService.addFileEntry(
+			RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+			_group.getGroupId(), DLFolderConstants.DEFAULT_PARENT_FOLDER_ID,
+			RandomTestUtil.randomString() + "." + ContentTypes.IMAGE_JPEG,
+			MimeTypesUtil.getExtensionContentType(ContentTypes.IMAGE_JPEG),
+			new byte[0], null, null,
+			ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+
+		MockHttpServletRequest mockHttpServletRequest =
+			new MockHttpServletRequest();
+
+		MockLiferayResourceRequest mockLiferayResourceRequest =
+			new MockLiferayResourceRequest();
+
+		mockLiferayResourceRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, _getThemeDisplay(mockHttpServletRequest));
+
+		mockHttpServletRequest.setAttribute(
+			JavaConstants.JAVAX_PORTLET_REQUEST, mockLiferayResourceRequest);
+
+		Assert.assertTrue(
+			_contentDashboardItemActionProvider.isShow(
+				_fileEntry, mockHttpServletRequest));
+	}
+
+	@Test
+	public void testIsShowWithDraftFileEntry() throws Exception {
+		User user = UserTestUtil.addUser();
+
+		try {
+			_fileEntry = _dlAppLocalService.addFileEntry(
+				RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+				_group.getGroupId(), DLFolderConstants.DEFAULT_PARENT_FOLDER_ID,
+				RandomTestUtil.randomString() + "." + ContentTypes.IMAGE_JPEG,
+				MimeTypesUtil.getExtensionContentType(ContentTypes.IMAGE_JPEG),
+				new byte[0], null, null,
+				ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+
+			DLFileEntry dlFileEntry = _dlFileEntryLocalService.getFileEntry(
+				_fileEntry.getFileEntryId());
+
+			DLFileVersion dlFileVersion = dlFileEntry.getFileVersion();
+
+			_dlFileEntryLocalService.updateStatus(
+				TestPropsValues.getUserId(), dlFileVersion.getFileVersionId(),
+				WorkflowConstants.STATUS_DRAFT,
+				ServiceContextTestUtil.getServiceContext(_group.getGroupId()),
+				new HashMap<String, Serializable>());
+
+			MockHttpServletRequest mockHttpServletRequest =
+				new MockHttpServletRequest();
+
+			ThemeDisplay themeDisplay = _getThemeDisplay(
+				mockHttpServletRequest);
+
+			themeDisplay.setPermissionChecker(
+				PermissionCheckerFactoryUtil.create(user));
+
+			MockLiferayResourceRequest mockLiferayResourceRequest =
+				new MockLiferayResourceRequest();
+
+			mockLiferayResourceRequest.setAttribute(
+				WebKeys.THEME_DISPLAY, themeDisplay);
+
+			mockHttpServletRequest.setAttribute(
+				JavaConstants.JAVAX_PORTLET_REQUEST,
+				mockLiferayResourceRequest);
+
+			Assert.assertTrue(
```

Comment:
I forgo @cgoncas, here que can use assertFalse instead of !_ right?

Reply:
This topic is controversial, because human brain read better the assertTrue with negative condition than the assertFalse, but I will change it because it is your :)